File: app/src/main/java/com/stoffe/gym/Database/ExerciseData.java
package com.stoffe.gym.Database;

import java.time.LocalDate;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class ExerciseData {

    public ExerciseData(int sets, int reps, int weight, int exerciseID) {
        this.sets = sets;
        this.reps = reps;
        this.weight = weight;
        this.exerciseID = exerciseID;
        date = LocalDate.now();
    }

    @PrimaryKey(autoGenerate = true)
    public int uid;

    public int exerciseID;

    @ColumnInfo(name = "sets")
    public int sets;

    @ColumnInfo(name = "reps")
    public int reps;

    @ColumnInfo(name = "weight")
    public int weight;

    @ColumnInfo(name = "date")
    public LocalDate date;

}

File: app/src/main/java/com/stoffe/gym/database/entities/ExerciseData.kt
package com.stoffe.gym.database.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDate

@Entity
data class ExerciseData(
    @PrimaryKey(autoGenerate = true) var uid: Int = 0,
    val sets: Int,
    val reps: Int,
    val weight: Float,
    val exerciseID: Int,
    val date: LocalDate = LocalDate.now(),
)
